test/e2e/e2e_test.go

```diff
@@ -108,54 +142,135 @@ var _ = Describe("controller", Ordered, func() {
+				}
+				fmt.Printf("pod %s running on %s\n", pods.Items[0].Name, pods.Items[0].Spec.NodeName)
+
+				return pods.Items[0].Spec.NodeName, nil
+			}
+			verifyControllerMgrPods := func() error {
+				_, e := verifyRunningPods(namespace, client.MatchingLabels{
+					"control-plane": "controller-manager",
+				}, 1)
+				return e
+			}
+			EventuallyWithOffset(1, verifyControllerMgrPods, time.Minute, time.Second).Should(Succeed())
+
+			By("validating that the nginx pod is running as expected")
+			var nodeName string
+			verifyNginxPods := func() error {
+				nodeName, err = verifyRunningPods("ingress-nginx", client.MatchingLabels{}, 1)
+				return err
+			}
+
+			EventuallyWithOffset(1,
+				verifyNginxPods,
+				time.Minute, time.Second).Should(Succeed())
+
+			//this is different than the eviction-autoscaler manager in that the pdb is generated
+			By("Verify PDB and PDBWatcher exist")
+			verifyPdbExists := func() error {
+				var pdbList = &policy.PodDisruptionBudgetList{}
+				err := clientset.List(ctx, pdbList, client.InNamespace("ingress-nginx"), client.Limit(1))
+				Expect(err).NotTo(HaveOccurred())
+				fmt.Printf("found %d pdbs in namespace ingress-nginx\n", len(pdbList.Items))
+				for _, pdb := range pdbList.Items {
+					fmt.Printf("found pdb name: %s \n", pdb.Name)
+					if pdb.Name != "ingress-nginx" {
+						return fmt.Errorf("nginx pdb is not present on cluster")
+					}
+					if pdb.Spec.MinAvailable != nil {
+						if val := pdb.Spec.MinAvailable.IntValue(); val != 1 {
+							return fmt.Errorf("PDB '%s' has MinAvailable set to %d (not 1)\n", pdb.Name, val)
+						}
+						fmt.Printf("PDB '%s' has MinAvailable set to 1\n", pdb.Name)
+					}
 				}
-				fmt.Printf("controller pod %s running on %s\n", pods.Items[0].Name, pods.Items[0].Spec.NodeName)
-				nodeName = pods.Items[0].Spec.NodeName
 				return nil
 			}
-			EventuallyWithOffset(1, verifyOneRunningPod, time.Minute, time.Second).Should(Succeed())
+			verifyEvictionAutoScalerExists := func() error {
+				var evictionAutoScalerList = &types.EvictionAutoScalerList{}
+				err = clientset.List(ctx, evictionAutoScalerList,
+					client.InNamespace("ingress-nginx"), &client.ListOptions{Limit: 1})
+				Expect(err).NotTo(HaveOccurred())
+				fmt.Printf("found %d evictionautoscalers in namespace ingress-nginx \n", len(evictionAutoScalerList.Items))
+				for _, resource := range evictionAutoScalerList.Items {
+					fmt.Printf("found evictionautoscaler name: %s \n", resource.GetName())
+					if resource.GetName() != "ingress-nginx" {
+						return fmt.Errorf("nginx evictionautoscalers is not present on cluster")
+					}
+					fmt.Printf("custom resource found: %s\n", resource.GetName())
+				}
+				return nil
+			}
+			EventuallyWithOffset(1, verifyPdbExists, time.Minute, time.Second).Should(Succeed())
+			EventuallyWithOffset(1, verifyEvictionAutoScalerExists, time.Minute, time.Second).Should(Succeed())
+
 			By("By Cordoning " + nodeName)
```

Comment:
should we validate we get > 2 pods after cordon?

Reply:
heh had this comment on 286 
//check that there are two pods temporarily or does that not matter as long as we successfully evicted?

gues it should jsut become a todo.